app.js

```diff
@@ -3,26 +3,59 @@
 const express = require('express');
 const app = express();
 const port = process.env.PORT || 3000;
-const playersData = require('');//should require the data.json file
+const playersData = require('./data.json'); //should require the data.json file
 app.use(express.json());
 
 app.get('/players', (req, res) => {
-    //should respond with the "players" array inside playersData and Status 200    
+  //should respond with the "players" array inside playersData and Status 200
+  res.send(playersData.players).status(200);
 });
 
 app.get('/players/:role', (req, res) => {
-    //should respond with only the players that have with the especified role. Status 200.
-    //If there's no player with the specified role it should respond with {"error": "No player found"} and Status 404.
+  //should respond with only the players that have with the especified role. Status 200.
+  //If there's no player with the specified role it should respond with {"error": "No player found"} and Status 404.
+  const playersByRole = playersData.players.filter((player) => {
+    return player.role === req.params.role;
+  });
+
+  playersByRole.length > 0
+    ? res.send(playersByRole).status(200)
+    : res.send({ error: 'No Player found' }).status(404);
```

Comment:
The .status() function should be executed before .send(). Your code returns status 200 OK and not 404.
Should be fixed with `res.status(404).send({ error: 'No Player found' });`.